### ISSUE TITLE
Error when parsing malformed query should raise an appropriate exception

### ISSUE THREAD

**Issue by such** [rmosolgo/graphql-ruby#3619]
**Describe the bug**

`NoMethodError` when parsing malformed query:

```
{
  "action": "execute",
  "controller": "graphql",
  "graphql": {
    "query": {
      "default": "query {\n  clubsReady {\n    name\n    slug\n    domesticLeague {\n      displayName\n      slug\n      country {\n        slug\n      }\n    }\n    country {\n      slug\n    }\n  }\n}"
    }
  },
  "query": {
    "default": "query {\n  clubsReady {\n    name\n    slug\n    domesticLeague {\n      displayName\n      slug\n      country {\n        slug\n      }\n    }\n    country {\n      slug\n    }\n  }\n}"
  }
}
```

**Versions**

`graphql` version: 1.12.16
`rails`: 6.1.4.1

**GraphQL query**

Example GraphQL query and response (if query execution is involved)

```graphql
"query": {
    "default": "query {\n  clubsReady {\n    name\n    slug\n    domesticLeague {\n      displayName\n      slug\n      country {\n        slug\n      }\n    }\n    country {\n      slug\n    }\n  }\n}"
  }
```

**Expected behavior**

Parsing error should be returned to the caller or a proper exception should be raised

**Actual behavior**

What specifically went wrong?

Place full backtrace here (if a Ruby exception is involved):

<details>
<summary>Click to view exception backtrace</summary>

```
undefined method `unpack' for #<ActionController::Parameters {"default"=>"query {\n  clubsReady {\n    name\n    slug\n    domesticLeague {\n      displayName\n      slug\n      country {\n        slug\n      }\n    }\n    country {\n      slug\n    }\n  }\n}"} permitted: false>


lib/graphql/language/lexer.rb:170 run_lexer |  
-- | --
lib/graphql/language/lexer.rb:11 tokenize |  
lib/graphql.rb:56 scan_with_ragel |  
lib/graphql.rb:52 scan |  
parser.y:452 block in parse_document |  
lib/graphql/tracing.rb:81 call_tracers |  
lib/graphql/tracing.rb:83 block in call_tracers |  
graphql-pro (1.18.2) lib/graphql/pro/operation_store/query_instrumentation.rb:19 trace |  
lib/graphql/tracing.rb:83 call_tracers |  
lib/graphql/tracing.rb:83 block in call_tracers |  
graphql-guard (2.0.0) lib/graphql/guard.rb:42 trace |  
lib/graphql/tracing.rb:83 call_tracers |  
lib/graphql/tracing.rb:83 block in call_tracers |  
lib/graphql/tracing/platform_tracing.rb:26 block in trace |  
lib/graphql/tracing/appsignal_tracing.rb:19 block in platform_trace |  
lib/appsignal/helpers/instrumentation.rb:525 instrument |  
lib/graphql/tracing/appsignal_tracing.rb:18 platform_trace |  
lib/graphql/tracing/platform_tracing.rb:25 trace |  
lib/graphql/tracing.rb:83 call_tracers |  
lib/graphql/tracing.rb:67 trace |  
parser.y:451 parse_document |  
parser.y:469 parse |  
lib/graphql.rb:47 parse_with_racc |  
lib/graphql.rb:35 parse |  
lib/graphql/query.rb:384 prepare_ast |  
lib/graphql/query.rb:445 with_prepared_ast |  
lib/graphql/query.rb:356 subscription? |  
lib/graphql/subscriptions/instrumentation.rb:24 before_query |  
lib/graphql/execution/instrumentation.rb:58 public_send |  
lib/graphql/execution/instrumentation.rb:58 block in call_hooks |  
lib/graphql/execution/instrumentation.rb:57 each |  
lib/graphql/execution/instrumentation.rb:57 call_hooks |  
lib/graphql/execution/instrumentation.rb:44 each_query_call_hooks |  
lib/graphql/execution/instrumentation.rb:27 block in apply_instrumenters |  
lib/graphql/execution/instrumentation.rb:72 call_hooks |  
lib/graphql/execution/instrumentation.rb:26 apply_instrumenters |  
lib/graphql/execution/multiplex.rb:187 instrument_and_analyze |  
lib/graphql/execution/multiplex.rb:62 block in run_queries |  
lib/graphql/tracing.rb:81 call_tracers |  
lib/graphql/tracing.rb:83 block in call_tracers |  
graphql-pro (1.18.2) lib/graphql/pro/operation_store/query_instrumentation.rb:19 trace |  
lib/graphql/tracing.rb:83 call_tracers |  
lib/graphql/tracing.rb:83 block in call_tracers |  
graphql-guard (2.0.0) lib/graphql/guard.rb:42 trace |  
lib/graphql/tracing.rb:83 call_tracers |  
lib/graphql/tracing.rb:83 block in call_tracers |  
lib/graphql/tracing/platform_tracing.rb:26 block in trace |  
lib/graphql/tracing/appsignal_tracing.rb:19 block in platform_trace |  
lib/appsignal/helpers/instrumentation.rb:525 instrument |  
lib/graphql/tracing/appsignal_tracing.rb:18 platform_trace |  
lib/graphql/tracing/platform_tracing.rb:25 trace |  
lib/graphql/tracing.rb:83 call_tracers |  
lib/graphql/tracing.rb:67 trace |  
lib/graphql/execution/multiplex.rb:60 run_queries |  
lib/graphql/execution/multiplex.rb:50 run_all |  
lib/graphql/schema.rb:1688 multiplex |  
lib/graphql/schema.rb:1659 execute |  
app/controllers/graphql_controller.rb:41 execute_query
```

</details>

**Additional context**

Our API is public and we want to be able to catch this error and display it to the caller.

**Comment by rmosolgo** [comment]
```
undefined method `unpack' for #<ActionController::Parameters ...>
```

Oh, so it received `ActionController::Parameters` instead of a string. Of course, graphql-ruby expects a string here. I think we can raise an argumenterror here instead, so that applications can rescue it as they see fit (a `422` or other 4xx response is probably most appropriate here for HTTP, since the incoming request had the wrong structure).

**Comment by such** [comment]
Yes that would be great!

**Comment by rmosolgo** [comment]
Fixed by #3628

**Comment by rmosolgo** [comment]
Thanks for the bump on this -- just shipped 1.12.17!
